src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -138,7 +138,33 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             else if (type == typeof(CodeBlockNodeModel))
             {
                 var code = obj["Code"].Value<string>();
-                node = new CodeBlockNodeModel(code, guid, 0.0, 0.0, libraryServices, ElementResolver);
+                CodeBlockNodeModel codeBlockNode = new CodeBlockNodeModel(code, guid, 0.0, 0.0, libraryServices, ElementResolver);
+                node = codeBlockNode;
+
+                // If the code block node is in an error state read the extra port data
+                // and initialize the input and output ports
+                if (node.IsInErrorState)
+                {
+                    List<string> inPortNames = new List<string>();
+                    var inputs = obj["Inputs"];
+                    foreach (var input in inputs)
+                    {
+                        inPortNames.Add(input["Name"].ToString());
+                    }
+
+                    List<int> outPortLineIndexes = new List<int>();
+                    var outputs = obj["Outputs"];
+                    foreach (var output in outputs)
+                    {
+                        // NOTE: This is parsing the line index from the description string
+                        //       It is not a great way to get the value, but we do not currently
+                        //       serialize the line indexes for ooutputs in code block nodes directly
+                        string lineIndexString = Regex.Match(output["Description"].ToString(), @"\d+").Value;
```

Comment:
what about simply incrementing this number, I don't think it's so important as the user will usually modify this codeblock and get it compiling soon after this state - at that point the indices will be recalculated.

I'm not really a fan of relying on the description containing the index (if you do want to keep this can you please add a sample of the description string to a comment here and do a null check etc.